docs/development/python-dependencies.rst

```diff
@@ -40,36 +30,57 @@ Pinning exact versions is important because it makes deployment
 predictable meaning that what you test and develop against locally is
 exactly reflected in production.
 
-Also, Socorro uses a `pip` wrapper called `peep`
-(https://pypi.python.org/pypi/peep) which ensures that the packages
-downloaded from the Python Package Index (PyPI) have not been tampered with.
+Also, Socorro uses ``pip>=8.0`` which has the ability to checksum
+check all dependencies so they are the exact same version we've
+verified and tested in local development.
 
-Since we can't trust peep to verify itself, we ship a version in the
-`./tools` directory of the Socorro repo.
+And to bootstrap ``pip``, we need a verified and vetted version of pip to boot,
+so we've included `./tools/pipstrap.py` (see
+https://github.com/erikrose/pipstrap) which makes sure we get that first
+``pip`` installed securely.
+
+The best tool to help you add all hashes needed for each package, is
+``hashin`` (https://github.com/peterbe/hashin). You can install this
+with ``pip install hashin``. This is something you only ever install
+in your local virtual environment.
 
 Whilst it's a given that you pin the exact version of the package you
 now depend on, that package might have its own dependencies and
 sometimes they're not pinned to specific version. For example,
-`web.py` depends on `somepackage` but doesn't state what exact
+``mypackage`` depends on ``somepackage`` but doesn't state what exact
 version. Therefore, it's your job to predict this before it's
-installed as a nested dependency. So, do this::
+installed as a nested dependency.
+
+The best approach is to inspect the ``setup.py`` of the new package(s)
```

Comment:
If I `somepackage==1.2.3` and its `setup.py` has a `install_requires=['depends==1.0.0']` will pip notice that too?